src/corona_hakab_model/consts.py

```diff
@@ -102,9 +71,42 @@
         lambda agent: agent.medical_state.name == "Recovered",
         lambda agent: agent.medical_state.name == "Symptomatic",
     ),  # TODO: Define better API
-    # dictionary of {date : percent} that controls what percentage of schools are open
-    "school_openage_factors": {1: 0, 30: 0.5, 45: 1},
-    "should_change_school_openage": False
+
+    # policies data
+    # a dictionary of type day:List[ConnectionTypes]. on each day, keeps only the given connection types opened
+    "policies_changes": {
+        40: ([ConnectionTypes.Family, ConnectionTypes.Other], "closing schools and works"),
+        70: ([ConnectionTypes.Family, ConnectionTypes.Other, ConnectionTypes.School], "opening schools"),
+        100: (ConnectionTypes, "opening works")
+    },
+    "change_policies": False,
+
+    # policies acting on a specific connection type, when a term is satisfied
+    "partial_opening_active": True,
+    # each connection type gets a list of conditioned policies
+    # each conditioned policy actives a specific policy when a condition is satisfied
+    # each policy changes the multiplication factor of a specific circle
+    # each policy is activated only if a list of terms is full-filled.
+    "connection_type_to_conditioned_policy": {
+        ConnectionTypes.School:
+        [
+            ConditionedPolicy(
+                activating_condition=lambda kwargs: len(np.flatnonzero(kwargs["manager"].contagiousness_vector)) > 1000,
+                policy=Policy(0, [lambda circle: random() > 0]), message="closing all schools"),
+            ConditionedPolicy(
+                activating_condition=lambda kwargs: len(np.flatnonzero(kwargs["manager"].contagiousness_vector)) < 500,
```

Comment:
so many lambdas in a single line is a code smell, use inheritance instead

Reply:
how will inheritance fix it?
the lambdas gives the researchers complete freedom without touching anything but consts, and without creating a hard-coded object / function for every different option.

Reply:
the researchers in this case are shahf and shir, and they both know how to code.
the contagiousness count in this case is only an example. there are many different possible activation conditions that they wanted to try out (for example, a given date, amount of agents in a given state, and so on).
i don't really see a way doing it without hard-coding an object for any possible condition.
any way, i agree that this whole module should be upgraded, and changed in the next version, but i think this is ok for now